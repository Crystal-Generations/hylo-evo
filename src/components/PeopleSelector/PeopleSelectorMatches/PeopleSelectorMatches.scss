.people-selector-matches {
  composes: messages-section from 'components/MessageSection/MessageSection.scss'
}
