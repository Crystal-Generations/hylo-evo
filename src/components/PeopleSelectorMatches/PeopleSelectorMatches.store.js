import { createSelector } from 'redux-orm'

import orm from 'store/models'

export const matchesSelector = createSelector(
  orm,
  state => state.orm,
  state => state.NewMessageThread,
  (session, thread) => {
    if (thread.autocomplete) {
      const term = thread.autocomplete.toLowerCase()
      const matches = session.Person
        .all()
        .filter(p => p.name.toLowerCase().includes(term))
        .orderBy('name')
        .toRefArray()
      if (matches.length > 0) {
        matches[0].active = true
      }
      return matches
    }
    return null
  }
)
