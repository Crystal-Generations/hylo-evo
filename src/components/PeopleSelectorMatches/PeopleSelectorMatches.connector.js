import { connect } from 'react-redux'

import { addMatch } from 'components/PeopleSelector/PeopleSelector.store'
import { matchesSelector } from './PeopleSelectorMatches.store'

export function mapStateToProps (state) {
  return {
    matches: matchesSelector(state)
  }
}

export default connect(mapStateToProps, { addMatch })
