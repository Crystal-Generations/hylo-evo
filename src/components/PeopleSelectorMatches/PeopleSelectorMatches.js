import React from 'react'

import './PeopleSelectorMatches.scss'

const { string } = React.PropTypes

export default function PeopleSelectorMatches () {
  return <div styleName='people-selector-matches'></div>
}

PeopleSelectorMatches.propTypes = {
  example: string
}
