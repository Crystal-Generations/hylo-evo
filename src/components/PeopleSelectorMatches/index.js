import PeopleSelectorMatches from './PeopleSelectorMatches'
import connector from './PeopleSelectorMatches.connector'
export default connector(PeopleSelectorMatches)
