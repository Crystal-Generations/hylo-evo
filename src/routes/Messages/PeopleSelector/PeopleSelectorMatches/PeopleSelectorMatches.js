import PropTypes from 'prop-types'
import React from 'react'
import PersonListItem from '../PersonListItem'
import './PeopleSelectorMatches.scss'

export default class PeopleSelectorMatches extends React.Component {
  render () {
    const { addParticipant, currentMatch, matches, setCurrentMatch } = this.props
    return <ul styleName='people-selector-matches'>
      {matches && matches.map(match =>
        <PersonListItem
          key={match.id}
          active={match.id === currentMatch.id}
          person={match}
          onClick={() => addParticipant(match)}
          onMouseOver={() => setCurrentMatch(match)} />)}
    </ul>
  }
}

PeopleSelectorMatches.propTypes = {
  addParticipant: PropTypes.func.isRequired,
  setCurrentMatch: PropTypes.func.isRequired,
  currentMatch: PropTypes.object,
  matches: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.any,
      name: PropTypes.string,
      avatarUrl: PropTypes.string,
      community: PropTypes.string
    })
  )
}
