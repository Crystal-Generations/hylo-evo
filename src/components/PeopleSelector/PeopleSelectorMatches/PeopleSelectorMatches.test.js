import { shallow } from 'enzyme'
import React from 'react'

import PeopleSelectorMatches from './PeopleSelectorMatches'

it('does something', () => {
  const matches = [
    {
      id: '1',
      name: 'Wombat',
      avatarUrl: 'https://flargle.com',
      community: 'Wombats'
    },
    {
      id: '2',
      name: 'Aardvark',
      avatarUrl: 'https://argle.com',
      community: 'Aardvarks'
    },
    {
      id: '3',
      name: 'Ocelot',
      avatarUrl: 'https://wargle.com',
      community: 'Ocelots'
    }
  ]
  const wrapper = shallow(<PeopleSelectorMatches matches={matches} />)
  expect(wrapper).toMatchSnapshot()
})
