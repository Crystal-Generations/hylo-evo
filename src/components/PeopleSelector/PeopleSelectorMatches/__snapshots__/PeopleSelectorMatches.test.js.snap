// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does something 1`] = `
<ul
  data-styleName="people-selector-matches"
>
  <PersonListItem
    active={false}
    onClick={[Function]}
    onMouseOver={[Function]}
    person={
      Object {
        "avatarUrl": "https://flargle.com",
        "community": "Wombats",
        "id": "1",
        "name": "Wombat",
      }
    }
  />
  <PersonListItem
    active={false}
    onClick={[Function]}
    onMouseOver={[Function]}
    person={
      Object {
        "avatarUrl": "https://argle.com",
        "community": "Aardvarks",
        "id": "2",
        "name": "Aardvark",
      }
    }
  />
  <PersonListItem
    active={false}
    onClick={[Function]}
    onMouseOver={[Function]}
    person={
      Object {
        "avatarUrl": "https://wargle.com",
        "community": "Ocelots",
        "id": "3",
        "name": "Ocelot",
      }
    }
  />
</ul>
`;
